clients/src/test/java/org/apache/kafka/common/network/SSLSelectorTest.java

```diff
@@ -218,6 +222,49 @@ public void testRenegotiation() throws Exception {
         }
     }
 
+
+    /**
+     * Tests handling of BUFFER_UNDERFLOW during unwrap when network read buffer is smaller than SSL session packet buffer size.
+     */
+    @Test
+    public void testNetReadBufferResize() throws Exception {
+        String node = "0";
+        createSelector(10, 0, 0);
```

Comment:
It is funny you should say that. I had started with a changeset to all the tests to create a selector with a helper method. And then reverted to this code because I thought it will be easier to get it reviewed with minimal changes to existing code :-) 

I am thinking of moving the new tests into a new test class SSLTransportLayerTest because I would like to add some more tests for SSL renegotiation as well. That would leave the selector tests sort of mirroring the plaintext selector tests and the tests specific to SSL transport layer in a separate class. Does that make sense?